org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -306,32 +310,48 @@ private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, Analys
       if (failedFiles.contains(resource)) {
         continue;
       }
-      List<Issue> rawIssues = resourceEntry.getValue();
-      try {
-        if (resource instanceof IFile) {
+      if (resource instanceof IFile) {
+        List<Issue> rawIssues = resourceEntry.getValue();
+        IPath path = resource.getFullPath();
+        try {
+          textFileBufferManager.connect(path, LocationKind.IFILE, new NullProgressMonitor());
+          ITextFileBuffer textFileBuffer = textFileBufferManager.getTextFileBuffer(path, LocationKind.IFILE);
+          IDocument document = textFileBuffer.getDocument();
+
           IProject project = resource.getProject();
           String moduleKey = project.getName();
           SonarLintCorePlugin.getDefault().getModulePathManager().setModulePath(moduleKey, project.getLocation().toString());
 
           String relativePath = resource.getProjectRelativePath().toString();
-          trackLocalIssues(moduleKey, resource, relativePath, rawIssues);
-          trackServerIssues(moduleKey, resource, relativePath);
-        } else {
-          // TODO delete if this never happens, or else handle it better
-          throw new IllegalStateException("updateMarkers for not an IFile?");
+          trackLocalIssues(moduleKey, relativePath, document, rawIssues);
+          trackServerIssues(moduleKey, relativePath, resource);
+        } finally {
+          textFileBufferManager.disconnect(path, LocationKind.IFILE, new NullProgressMonitor());
         }
-      } catch (Exception e) {
-        SonarLintCorePlugin.getDefault().error("Unable to compute position of SonarLint marker on resource " + resource.getName(), e);
+      } else {
+        // TODO handle non-file-level issues
       }
     }
   }
 
-  private void trackLocalIssues(String moduleKey, IResource resource, String relativePath, List<Issue> rawIssues) {
-    Collection<MutableTrackable> trackables = rawIssues.stream().map(IssueTrackable::new).collect(Collectors.toList());
+  private static void trackLocalIssues(String moduleKey, String relativePath, IDocument document, List<Issue> rawIssues) {
+    List<MutableTrackable> trackables = rawIssues.stream().map(issue -> {
+      TextRange textRange = new TextRange(issue.getStartLine(), issue.getStartLineOffset(), issue.getEndLine(), issue.getEndLineOffset());
+      String content = null;
+      FlatTextRange flatTextRange = MarkerUtils.toFlatTextRange(document, textRange);
```

Comment:
Minor things: 
Why not have a `MarkerUtils#getFlatTextRange(Document, Issue)` and remove the intermediate textRange ?
I also don't like too much #**to**`FlatTextRange` because it feels like it's a trivial conversion, when in fact it is not and can return null.